cmd2/rl_utils.py

```diff
@@ -0,0 +1,66 @@
+# coding=utf-8
+"""
+Imports the proper readline for the platform and provides utility functions for it
+"""
+import sys
+
+try:
+    from enum34 import Enum
+except ImportError:
+    from enum import Enum
+
+# Prefer statically linked gnureadline if available (for macOS compatibility due to issues with libedit)
+try:
+    import gnureadline as readline
+except ImportError:
+    # Try to import readline, but allow failure for convenience in Windows unit testing
+    # Note: If this actually fails, you should install readline on Linux or Mac or pyreadline on Windows
+    try:
+        # noinspection PyUnresolvedReferences
+        import readline
+    except ImportError:  # pragma: no cover
+        pass
+
+
+class RlType(Enum):
+    """Readline library types we recognize"""
+    GNU = 1
+    PYREADLINE = 2
+    NONE = 3
+
+
+# Check what implementation of readline we are using
+
+rl_type = RlType.NONE
+
+# The order of this check matters since importing pyreadline will also show readline in the modules list
+if 'pyreadline' in sys.modules:
+    rl_type = RlType.PYREADLINE
+
+elif 'gnureadline' in sys.modules or 'readline' in sys.modules:
+    rl_type = RlType.GNU
+
+    # Load the readline lib so we can access members of it
+    import ctypes
+    readline_lib = ctypes.CDLL(readline.__file__)
+
+
+def rl_force_redisplay() -> None:
+    """
+    Causes readline to redraw prompt and input line
+    """
+    if not sys.stdout.isatty():
+        return
+
+    if rl_type == RlType.GNU:  # pragma: no cover
```

Comment:
This is covered on macOS when the ``gnureadline`` module is installed.

Reply:
The reason I blocked this out is because it can't be covered  by the automated tests because there's isn't a terminal.

Reply:
I guess it depends on where we care about seeing the coverage.

Reply:
Oh good point.  I think it is reasonable to block it out if we know it won't be hit by any of our CI systems.